src/components/FavoriteList/FavoriteList.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { Title, Content, FavoriteList, FilterBar } from "./FavoriteList.styled";
+import favorites from '../../actions/favorites';
+import GifThumb from "../GifThumb";
+import SVG from "../SVG";
+
+export class FavoriteListComponent extends React.Component {
+	constructor(props) {
+		super(props);
+		this.input = null;
+		this.filter = this.filter.bind(this);
+		this.onKeyDown = this.onKeyDown.bind(this);
+	}
+	filter() {
+		this.props.filterFavorite(this.input.value);
+		return this.input.value;
+	}
+	onKeyDown(event) {
+		if (event.keyCode === 13)
+		{
+			this.filter();
+		}
+	}
+	componentWillUnmount()
+	{
+		this.props.filterFavorite('');
+	}
+	render() {
+		var self = this;
```

Comment:
not regarding react, but avoid using `var`, as `let` and `const` are scoped by block and `var` is scoped to function (so, buggier)